### PR TITLE
{RDBMS} Introduce tests for Vnet/Subnet and public-access in Flexible-server commands. 

### PR DESCRIPTION
**Pull request by arde0708**
**Description** 
The PR introduces tests for vnet, subnet and public-access for flexible-server commands.

**Testing Guide**  
azdev test FlexibleServerVnetMgmtScenarioTest
azdev test FlexibleServerPublicAccessMgmtScenarioTest --live

### CODE REVIEW

**Comment by yonzhan** [comment]
RDBMS

File: src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py [hunk 1]
```diff
@@ -467,3 +468,448 @@ def _test_flexible_server_replica_mgmt(self, database_engine, resource_group):
...
+        vnet_name_2 = 'clitestvnet1'
+        subnet_name_2 = 'clitestsubnet1'
+        # flexible-servers
+        servers = [self.create_random_name(SERVER_NAME_PREFIX, SERVER_NAME_MAX_LENGTH),
+                   self.create_random_name('azuredbcli', SERVER_NAME_MAX_LENGTH)]
+
+        # Case 1 : Provision a server with supplied Subnet ID that exists, where the subnet is not delegated
+
+        # create vnet and subnet
+        vnet_result = self.cmd('network vnet create -n {} -g {} -l {} --address-prefix {} --subnet-name {} --subnet-prefix {}'
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py:560]: Could you use VirtualNetworkPreparer() as shown here https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/dls/tests/latest/test_dls_commands.py#L308?

File: src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py [hunk 2]
```diff
@@ -467,3 +468,448 @@ def _test_flexible_server_replica_mgmt(self, database_engine, resource_group):
...
+        self.cmd('network vnet subnet update -g {} --name {} --vnet-name {} --remove delegations'.format(resource_group,
+                                                                                                         subnet_name_1,
+                                                                                                         vnet_name))
+        self.cmd('network vnet subnet update -g {} --name {} --vnet-name {} --remove delegations'.format(resource_group,
+                                                                                                         'Subnet' + servers[1][6:],
+                                                                                                         vnet_name_2))
+
+        # remove all vnets
+        self.cmd('network vnet delete -g {} -n {}'.format(resource_group, vnet_name))
+        self.cmd('network vnet delete -g {} -n {}'.format(resource_group, vnet_name_2))
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py:610]: With VirtualNetworkPreparer, this part is not needed.

File: src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py [hunk 3]
```diff
@@ -467,3 +468,448 @@ def _test_flexible_server_replica_mgmt(self, database_engine, resource_group):
...
+        vnet_name_2 = 'clitestvnet3'
+
+        # flexible-servers
+        servers = [self.create_random_name('azuredbvname1', SERVER_NAME_MAX_LENGTH),
+                   self.create_random_name('azuredbvname2', SERVER_NAME_MAX_LENGTH)]
+
+        # Case 1 : Provision a server with supplied Vname that exists.
+
+        # create vnet and subnet. When vnet name is supplied, the subnet created will be given the default name.
+        vnet_result = self.cmd('network vnet create -n {} -g {} -l {} --address-prefix {} --subnet-name {} --subnet-prefix {}'
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py:637]: same as above

File: src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py [hunk 4]
```diff
@@ -467,3 +468,448 @@ def _test_flexible_server_replica_mgmt(self, database_engine, resource_group):
...
+        vnet_name_2 = 'clitestvnet6'
+
+        # flexible-servers
+        servers = [self.create_random_name('azuredbvname3', SERVER_NAME_MAX_LENGTH),
+                   self.create_random_name('azuredbvname4', SERVER_NAME_MAX_LENGTH)]
+
+        # Case 1 : Provision a server with supplied Vname and subnet name that exists.
+
+        # create vnet and subnet. When vnet name is supplied, the subnet created will be given the default name.
+        vnet_result = self.cmd('network vnet create -n {} -g {} -l {} --address-prefix {} --subnet-name {} --subnet-prefix {}'
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py:719]: same

File: src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py [hunk 5]
```diff
@@ -467,3 +468,448 @@ def _test_flexible_server_replica_mgmt(self, database_engine, resource_group):
...
+    '''
+    @AllowLargeResponse()
+    @ResourceGroupPreparer(location=mysql_location)
+    def test_investigation(self, resource_group):
+        self.helper('mysql', resource_group)
+
+    '''
+    @AllowLargeResponse()
+    @ResourceGroupPreparer(location=postgres_location)
+    @live_only()
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py:486]: could you remove @live_only() and let me see the error in CI system when playback?

File: src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py (hunk 2 above)
Comment (arde0708) [reply src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py:610]: In our case - there is an additional step before the vnets can be deleted which is to remove the delegations( Line 667 above). Will try and see if that is taken care off if we use the VirtualNetworkPreparer()

Comment (arde0708) [reply src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py:610]: > With VirtualNetworkPreparer, this part is not needed.

In our case - there is an additional step before the vnets can be deleted which is to remove the delegations( Line 667 above). Will try and see if that is taken care off if we use the VirtualNetworkPreparer()

File: src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py (hunk 5 above)
Comment (arde0708) [reply src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py:495]: > could you remove @live_only() and let me see the error in CI system when playback?

@Juliehzl : Removed. The test test_investigation2() is the concerning test. I have pinged you offline about the details.

File: src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py (hunk 1 above)
Comment (arde0708) [reply src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py:547]: Ended up removing all the dependencies for live-only .

File: src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py (hunk 5 above)
Comment (arde0708) [reply src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py]: Was able to fix the open issues. Hence, removed the tests I had added for investigation.

**Comment by arde0708** [comment]
I had to re raise this PR as this got messed up. Please consider reviewing this PR instead : https://github.com/Azure/azure-cli/pull/15830

**Pull request closed without merging**